app/forms/declaration/DeclarantDetails.scala

```diff
@@ -25,7 +27,15 @@ object DeclarantDetails {
 
   val id = "DeclarantDetails"
 
-  val mapping = Forms.mapping("details" -> EntityDetails.mapping)(DeclarantDetails.apply)(DeclarantDetails.unapply)
+  val declarantMapping = Forms
+    .mapping(
+      "eori" ->
+        text()
+          .verifying("supplementary.eori.empty", nonEmpty)
+          .verifying("supplementary.eori.error", noLongerThan(17) and isAlphanumeric)
+    )((eori => EntityDetails(Some(eori), None)))(entityDetails => entityDetails.eori)
```

Comment:
This part is little tricky. I think it will be better to prepare new class for DeclarantDetails page. We don't need to reuse old component. Now this page doesn't have address, so in the future it will be difficult to read, why we changed this or go with this way

Reply:
As I mentioned last week, there is some discussion about whether this is the final solution for this problem.  We might end up removing this page altogether and extracting the EORI number of the declarant from the logged in user details.  

I therefore decided that changing the model (and the BE) wasn't worth the effort.

Reply:
I don't think we should couple our model to our form structure. I know it already is but let's not make it worse.

I think what @SteveSugden has done is fine although could maybe be done like:
`(EntityDetails.apply)(_.eori)`
You'd need a separate apply method that just accepts an `Option[String]` though